packages/protocol/contracts/identity/FederatedAttestations.sol

```diff
@@ -598,62 +330,69 @@ contract FederatedAttestations is
     address signer,
+    uint64 issuedOn,
     uint8 v,
     bytes32 r,
     bytes32 s
-  ) public isValidUser(issuer, account) {
-    require(
-      isValidAttestation(identifier, issuer, account, issuedOn, signer, v, r, s),
-      "Signature is invalid"
-    );
+  ) external {
+    // TODO allow for updating existing attestation by only updating signer/publishedOn/issuedOn
+    validateAttestation(identifier, issuer, account, issuedOn, signer, v, r, s);
     for (uint256 i = 0; i < identifierToAttestations[identifier][issuer].length; i = i.add(1)) {
       // This enforces only one attestation to be uploaded
       // for a given set of (identifier, issuer, account)
-      // Editing/upgrading an attestation requires that it be deleted before a new one is registered
+      // Editing/upgrading an attestation requires that it be revoked before a new one is registered
       require(
         identifierToAttestations[identifier][issuer][i].account != account,
         "Attestation for this account already exists"
       );
     }
-    OwnershipAttestation memory attestation = OwnershipAttestation(account, issuedOn, signer);
+    uint64 publishedOn = uint64(block.timestamp);
+    OwnershipAttestation memory attestation = OwnershipAttestation(
+      account,
+      signer,
+      issuedOn,
+      publishedOn
+    );
     identifierToAttestations[identifier][issuer].push(attestation);
     addressToIdentifiers[account][issuer].push(identifier);
-    emit AttestationRegistered(identifier, issuer, account, issuedOn, signer);
+    emit AttestationRegistered(identifier, issuer, account, signer, issuedOn, publishedOn);
   }
 
   /**
-   * @notice Deletes an attestation 
-   * @param identifier Hash of the identifier to be deleted
+   * @notice Revokes an attestation 
+   * @param identifier Hash of the identifier to be revoked
    * @param issuer Address of the attestation issuer
    * @param account Address of the account mapped to the identifier
-   * @dev Throws if sender is not the issuer, account, or an authorized AttestationSigner
+   * @dev Throws if sender is not the issuer, signer, or account
    */
-  function deleteAttestation(bytes32 identifier, address issuer, address account)
-    public
-    isValidUser(issuer, account)
-  {
+  // TODO should we pass in the issuedOn/signer parameter? ie. only revoke if the sender knows
+  // the issuedOn/signer for the unique attestation
+  function revokeAttestation(bytes32 identifier, address issuer, address account) external {
     OwnershipAttestation[] memory attestations = identifierToAttestations[identifier][issuer];
     for (uint256 i = 0; i < attestations.length; i = i.add(1)) {
       OwnershipAttestation memory attestation = attestations[i];
       if (attestation.account == account) {
+        address signer = attestation.signer;
+        uint64 issuedOn = attestation.issuedOn;
+        require(
+          signer == msg.sender || issuer == msg.sender || account == msg.sender,
```

Comment:
If we do want to do that, it may be worth adding a few test cases for that (previous attestation signer that is removed, and then tries to register/whatever) 🤔

Reply:
That makes sense! Also I think raises the question: do we want to allow the signer of the attestation to revoke if they have been removed as a signer?

Reply:
> do we want to allow the signer of the attestation to revoke if they have been removed as a signer?
good question, allowing this would allow them to revoke previous, legitimate attestations but restricted to only those issued by this particular signer; I tend towards no on this. That said, if we allow any current registered signer to revoke attestations from an issuer, that also makes it possible for them to revoke attestations until the breach is discovered and the signer is deregistered. We should probably include a comment about this latter tradeoff + make sure to get a second opinion on this from the reviewers?